src/ddscxx/include/org/eclipse/cyclonedds/topic/datatopic.hpp

```diff
@@ -329,7 +345,7 @@ ddsi_serdata *serdata_from_sample(
   d->resize(sz);
   ptr = static_cast<unsigned char*>(d->data());
   memset(ptr, 0x0, 4);
-  if (str.local_endianness() == org::eclipse::cyclonedds::core::cdr::endianness::little_endian)
+  if (native_endianness() == endianness::little_endian)
     *(ptr + 1) = 0x1;
 
   str.set_buffer(calc_offset(d->data(), 4));
```

Comment:
Shouldn't we also call `write()` based on the endianness? (with in the scope of the current changes, assuming endianness is not part of the streamer functions)

```diff
  switch (kind)
  {
  case SDK_KEY:
    key_write(str, msg);
    break;
  case SDK_DATA:
-     write(str, msg);
+    if (native_endianness() == endianness::little_endian)
+      write(str, msg);
+    else
+      write_swapped(str, msg);
    break;
  case SDK_EMPTY:
    assert(0);
  }
```

Reply:
Probably this is also true for `move()`, though I think `move()` and `move_swapped()` should be same

Reply:
There is a need for that (assuming I am not overlooking something): while there are some places in the spec where byte order in the CDR stream is mandated (like in computing a DDSI keyhash as well as in some places in DDS Security — and perhaps elsewhere) for normal application data, there is no required byte order. Always serializing to native byte order therefore generally makes the most sense.